tests/integration/test_scale.py

```diff
@@ -0,0 +1,173 @@
+        machines: The machines to deploy the any-charm onto
+        entries: List of DNS entries for any-charm
+    """
+    anyapp_name = f"anyapp-t{any_app_number}"
+    if machines is not None:
+        machine = machines[any_app_number % len(machines)]
+    else:
+        machine = None
+    logger.info("Deploying %s on %s", anyapp_name, machine)
+    if anyapp_name in model.applications:
+        return
+    await tests.integration.helpers.generate_anycharm_relation(
+        app,
+        ops_test,
+        anyapp_name,
+        entries,
+        machine=machine,
+    )
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+@pytest.mark.skip(reason="Scaling test")
+async def test_lots_of_applications(
+    app: ops.model.Application,
+    ops_test: OpsTest,
+    model: Model,
+):
+    """
+    arrange: build and deploy the charm.
+    act: deploy a lot of applications and relate them to the bind operator
+    assert:
+    """
+    batch_number = 10
+    machines_number = 20
+    status = await ops_test.juju("status", "--format", "json")
+    data = json.loads(status[1])
+    machines_available = (
+        sum(
+            1
+            for machine in data["machines"].values()
+            if machine["machine-status"]["current"] == "running"
+        )
+        - 1
+    )
+    while machines_available < machines_number:
+        await ops_test.juju("add-machine")
+        status = await ops_test.juju("status", "--format", "json")
+        data = json.loads(status[1])
+        machines_available = (
+            sum(
+                1
+                for machine in data["machines"].values()
+                if machine["machine-status"]["current"] == "running"
+            )
+            - 1
+        )
+        print("Available machines:", machines_available)
+        time.sleep(10)
+
+    for i in range(int(2000 / batch_number)):
+
+        status = await ops_test.juju("status", "--format", "json")
+        status_data = json.loads(status[1])
+        # we collect the machines but leave out machine "0"
+        # that should be in use by the bind-operator
+        machines = [x for x in status_data["machines"].keys() if x != "0"]
+        print("Available machines:", machines)
+
+        await asyncio.gather(
+            *[
+                deploy_any_charm(
+                    app,
+                    ops_test,
+                    model,
+                    i * batch_number + x,
+                    machines,
+                    [
+                        models.DnsEntry(
+                            domain="dns.test",
+                            host_label=f"admin{i * batch_number + x}",
+                            ttl=5,
+                            record_class="IN",
+                            record_type="A",
+                            record_data=(
+                                "1.1."
+                                f"{int(i * batch_number + x / 255) + 1}."
```

Comment:
maybe % instead of / ?, something like `f"{int((i * batch_number + x) % 255) + 1}."`

Reply:
I forgot the parenthesis but it is a division that I want, otherwise it would be the same value as the next line. What I'm doing here is basically counting in base 255